From 7eb333fbdde32dbebdcc88c35610159e207237c9 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Tue, 14 Jul 2015 14:57:55 -0700
Subject: [PATCH] Linux 4.2 compat: remove bio->bi_cnt access

Linux 4.2 commit torvalds/linux@dac5621 renamed bio->bi_cnt to
bio->__bi_cnt.  Because this value is only used once in a block of
debug code it simplest just to remove the PANIC.  To my knowledge
this debugging has never been hit or proved useful so this is no
great loss.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Richard Yao <ryao@gentoo.org>
Closes #3596
---
 module/zfs/vdev_disk.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/module/zfs/vdev_disk.c b/module/zfs/vdev_disk.c
index 9cfc080..eb77c26 100644
--- a/module/zfs/vdev_disk.c
+++ b/module/zfs/vdev_disk.c
@@ -426,15 +426,6 @@ BIO_END_IO_PROTO(vdev_disk_physio_completion, bio, size, error)
 	dio_request_t *dr = bio->bi_private;
 	int rc;
 
-	/* Fatal error but print some useful debugging before asserting */
-	if (dr == NULL)
-		PANIC("dr == NULL, bio->bi_private == NULL\n"
-		    "bi_next: %p, bi_flags: %lx, bi_rw: %lu, bi_vcnt: %d\n"
-		    "bi_idx: %d, bi_size: %d, bi_end_io: %p, bi_cnt: %d\n",
-		    bio->bi_next, bio->bi_flags, bio->bi_rw, bio->bi_vcnt,
-		    BIO_BI_IDX(bio), BIO_BI_SIZE(bio), bio->bi_end_io,
-		    atomic_read(&bio->bi_cnt));
-
 #ifndef HAVE_2ARGS_BIO_END_IO_T
 	if (BIO_BI_SIZE(bio))
 		return (1);
